misc-scripts/gifts_import/populate_protein_feature.pl

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env perl
+# Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+# Copyright [2016-2018] EMBL-European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#The script populates the CORE db's protein_feature table with all the uniprot mappings found in GIFTS db's ensp_u_cigar table.
+
+
+use strict;
+use warnings;
+use DBI qw( :sql_types );
+use Getopt::Long;
+use Bio::EnsEMBL::Registry;
+use Data::Dumper;
+
+my ( $gfhost, $gfport, $gfuser, $gfpass, $gfdb );
+my ( $pfhost, $pfport, $pfuser, $pfpass, $pfdb );
+
+my $create_index = 0;
+my $dryrun = 0;
+#my $limit = 10;
+my $limit = 0;
+
+
+GetOptions(
+  "gfhost=s"    => \$gfhost,
+  "gfport=i"       => \$gfport,
+  "gfuser=s"    => \$gfuser,
+  "gfpass=s"    => \$gfpass,
+  "gfdb=s"   => \$gfdb,
+
+  "pfhost=s"      => \$pfhost,
+  "pfport=i"      => \$pfport,
+  "pfuser=s"      => \$pfuser,
+  "pfpass=s"        => \$pfpass,
+  "pfdb=s"   => \$pfdb,
+
+  "dryrun!"         => \$dryrun,
+  "help", \&usage,
+
+);
+
+usage() if ( !defined $gfhost || !defined $gfport || !defined $gfuser || !defined $gfdb );
+usage() if ( !defined $pfhost  || !defined $pfport || !defined $pfuser || !defined $pfpass );
+
+
+my ( $giftsDB, $protfeatureDB ) = init_db();
+die unless defined($giftsDB) and defined($protfeatureDB);
+
+print "giftsDB credentials:\n";
+print Dumper($giftsDB);
+
+print "protfeatureDB credentials:\n";
+print Dumper($protfeatureDB);
+
+
+my $registry = get_registry();
+my $translation_adaptor = $registry->get_adaptor( 'human', 'core', 'translation' );
+
+
+sub init_db {
+
+# This is where the gifts info will be read from
+  my $giftsDB = {
+    user => $gfuser,
+    pass => $gfpass,
+    host => $gfhost,
+    port => $gfport,
+    dbname => $gfdb
+
+  };
+
+# This is where the gifts info will be written to
+  my $protfeatureDB = {
+    user   => $pfuser,
+    pass   => $pfpass,
+    host   => $pfhost,
+    port   => $pfport,
+    dbname => $pfdb
+  };
+
+  return ( $giftsDB, $protfeatureDB );
+}
+
+
+populate_protein_feature_db();
```

Comment:
Could this be changed not to intersperse script logic with subroutine declarations please.